stock_request_mrp/hooks.py

```diff
@@ -0,0 +1,70 @@
+# Copyright 2020 ForgeFlow S.L. (https://www.forgeflow.com)
+# License LGPL-3.0 or later (https://www.gnu.org/licenses/lgpl.html).
+
+import logging
+
+from odoo import SUPERUSER_ID, api
+
+logger = logging.getLogger(__name__)
+
+
+def post_init_hook(cr, registry):
+    """
+    The objective of this hook is to link existing MOs
+    coming from a Stock Request.
+    """
+    logger.info("Linking existing MOs coming from a Stock Request")
+    link_existing_mos_to_stock_request(cr)
+
+
+def link_existing_mos_to_stock_request(cr):
+    env = api.Environment(cr, SUPERUSER_ID, dict())
+    stock_request_obj = env["stock.request"]
+    stock_request_order_obj = env["stock.request.order"]
+    stock_request_allocation_obj = env["stock.request.allocation"]
+    mrp_production_obj = env["mrp.production"]
+    mos_with_sr = mrp_production_obj.search([("origin", "ilike", "SR/%")])
+    logger.info("Linking %s MOs records" % len(mos_with_sr))
+    for mo in mos_with_sr:
+        stock_request = stock_request_obj.search([("name", "=", mo.origin)])
```

Comment:
Have you tested migration with a lot of production orders and requests ? I would have done search outside loop

Reply:
I moved search outside loop as you request, it seems slightly faster :)